src/components/SignatureRequest.js

```diff
@@ -0,0 +1,182 @@
+
+  const wallet = useWallet()
+  const theme = useTheme()
+
+  const illustration = getIllustration(signingError, signHash)
+  const { statusText, statusTextColor } = getStatusText(
+    signingError,
+    signHash,
+    theme
+  )
+  const infoText = getInfoText(signingError, signHash, successText, actionText)
+
+  const buttonWidth = compactMode ? '100% ' : `calc((100% - ${2 * GU}px) /  2)`
+
+  useEffect(() => {
+    const requestSignature = async () => {
+      if (!wallet || signingError) {
+        return
+      }
+      const now = dayjs().toString()
+      const { signHash, error } = await signMessage(wallet, now)
+
+      if (error) {
+        setSigningError(true)
+        return
+      }
+      const { error: actionError } = await onSignSuccessAction()
+
+      if (actionError) {
+        onActionErrorScreen()
+        return
+      }
+
+      setSignHash(signHash)
+
+      if (onActionSuccessScreen) {
+        const timer = setTimeout(() => {
+          onActionSuccessScreen()
+        }, 3000)
+        return () => clearTimeout(timer)
+      }
+    }
+    requestSignature()
+
+    /* We only need to execute it at the beginning but in order to be able to execute it again
+     if the repeat signature button is clicked we need the signingError as a dependency */
+  }, [signingError]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  return (
+    <div
+      css={`
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        padding: ${5 * GU}px;
+      `}
+    >
+      <div
+        css={`
+          display: flex;
+          flex-direction: column;
+          text-align: center;
+        `}
+      >
+        <img src={illustration} />
+        <span
+          css={`
+            ${textStyle('title2')};
+            margin-top: ${4 * GU}px;
+          `}
+        >
+          {title}
+        </span>
+
+        <span
+          css={`
+            ${textStyle('body2')};
+            color: ${statusTextColor};
+            margin-top: ${1.5 * GU}px;
+          `}
+        >
+          {statusText}
+        </span>
+        <span
+          css={`
+            ${textStyle('body2')};
+            color: ${theme.surfaceContentSecondary};
+            margin-top: ${1.5 * GU}px;
+          `}
+        >
+          {infoText}
+        </span>
+        {signingError && (
+          <div
+            css={`
+              display: flex;
+              justify-content: space-between;
+              width: 100%;
+              flex-direction: ${compactMode ? 'column' : 'row'};
+              margin-bottom: ${1.5 * GU}px;
+              margin-top: ${3 * GU}px;
+            `}
+          >
+            <ActionButtons onClick={() => {}} width={buttonWidth}>
+              Troubleshooting
+            </ActionButtons>
+            <ActionButtons
+              mode="strong"
+              onClick={() => {
+                setSigningError(false)
+              }}
```

Comment:
Maybe we could have a `useCallback` above (`handleRepeatSignature` or similar) to prevent re creating the function on every render?

Reply:
could you provide an example? not sure if i follow the idea

Reply:
Havinig like

```
const handleRepeatSignature = useCallback(() => setSigningError(false))
```

And passing this function to the `onClick`